### PR TITLE
Add support for format in pivot tables

### PR DESCRIPTION
**Pull request by stonefield**
Support number format in pivot tables.

### CODE REVIEW

**Comment by stonefield** [comment]
@kiskoza How about merging this in. Easy fix, low risk.

**Comment by kiskoza** [comment]
hi @stonefield . I don't have a merge permission on this repo, I'm just helping people here and there. I'm not sure why was this PR set aside, but here's my opinion:

Your code change in the lib seems to be right. However I don't like the idea of removing a test to prove a new feature, so your test code change should be moved to a new test case instead of changing an existing one. Unfortunately I never used pivot tables, so it would help a lot if you attach some screenshots / code snippets about the new features (what it looked like before your code change, what it looks like with this PR, how to achieve the old look if somebody wants that).

If you have any further questions, I'm here to help or you can [join our Slack](https://join.slack.com/t/caxlsx/shared_invite/enQtOTI5OTM0MzI1Njk5LTBlMDQzNDk2YzkwODMxMmVkODMyYzJiZGU5NTQ3YTg5NTBlN2IwZTlmNTRjNzhiY2E0MDY2OTEyYmFlODI5NjA)

**Comment by stonefield** [comment]
Thanks for the feedback @kiskoza 
I will extract the test and provide an example on the difference it makes next week.

**Comment by stonefield** [comment]
@kiskoza I have separated the test. I have also made an example of what formatting means to a pivot table.
The enclosed spreadsheet shows both formatted and unformatted pivot tables, where you can see the difference. 
Who should I ask to make the merge?
[PivotTableExample.xlsx](https://github.com/caxlsx/caxlsx/files/6134774/PivotTableExample.xlsx)

**Comment by kiskoza** [comment]
Thanks for the test changes and the example file, now I see what you tries to accomplish - all looks good to me.

@noniq could you check and merge?

File: test/workbook/worksheet/tc_pivot_table.rb [hunk 1]
```diff
@@ -132,4 +132,11 @@ def test_to_xml_string_with_options_on_data_field
     end
     shared_test_pivot_table_xml_validity(pivot_table)
   end
+
+  def test_to_xml_string_with_format_options_on_data_field
+    pivot_table = @ws.add_pivot_table('G5:G6', 'A1:E5') do |pt|
+      pt.data = [{:ref=>"Sales", :subtotal => 'sum', num_fmt: 4}]
+    end
+    shared_test_pivot_table_xml_validity(pivot_table)
+  end
```
Comment (noniq) [inline test/workbook/worksheet/tc_pivot_table.rb:141]: Please add an assertion to also make sure that the `num_fmt` code actually ends up in the XML (currently the test still passes even if line 219 in pivot_table.rb is removed).

**Review by noniq — CHANGES_REQUESTED** [review]
Looks great, almost ready for merging! Please address the minor issue regarding the test case.

File: test/workbook/worksheet/tc_pivot_table.rb (hunk 1 above)
Comment (stonefield) [reply test/workbook/worksheet/tc_pivot_table.rb:142]: I made an improvement to the test. I hope that represents the change you pointed out should be there.

**Review by noniq — APPROVED** [review]

**Comment by noniq** [comment]
Perfect. Thanks for your contribution!

**Pull request merged**